code/datums/ai_laws.dm

```diff
@@ -5,16 +5,48 @@
 #define LAW_ION "ion"
 
 
+// ai_laws holds lists of laws
+// Each law has a description of actual law telling the player what to do
+// and also has a state for storing whether the law should be stated pubicly or not
+// state can be changed the the player
+
+// description is the actual law
+// state is whether the law should be stated publicly or not (AI's choice)
+/datum/ai_law
+	var/description = null
+	var/state = null

```

Comment:
Two things. One, description is a string and should be var/description = "" and since state is a bool, it should be var/state = FALSE instead of null. This gives future coders a bit of a hint as to what the values of these ought to be.

Two, state is probably the most generic name imaginable. Could you change it to "announce" or something that actually describes what it does better than state?